src/Adapter/Address/CommandHandler/SetRequiredFieldsForAddressHandler.php

```diff
@@ -46,7 +46,7 @@ final class SetRequiredFieldsForAddressHandler implements SetRequiredFieldsForAd
      */
     public function handle(SetRequiredFieldsForAddressCommand $command)
     {
-        $address = new Address();
+        $address = new CustomerAddress();
```

Comment:
One question, how does it work for brand or manufacturer address? Because I believe they all share the same `Address` class But using `CustomerAddress ` here means it's only dedicated to customer addresses, but not the other ones? Then shouldn't the handler name and namespace reflect this limitation?
=> `SetRequiredFieldsForCustomerAddressHandler::handle(SetRequiredFieldsForCustomerAddressCommand $command)`

Reply:
Only customer address is affected :man_shrugging:
Maybe as you say it could be reworked in 178.
It came from legacy that in db required_fields are saved by classname of the object (in this case CustomerAddress. By providing Address it doesn't work at all).